src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs

```diff
@@ -79,14 +55,50 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
             {
                 int errorCode = Marshal.GetLastWin32Error();
 
-                // Handle the possible race condition of someone else connecting to the server 
-                // between our calls to WaitNamedPipe & CreateFile.
-                if (errorCode == Interop.Errors.ERROR_PIPE_BUSY)
+                if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND ||
```

Comment:
Can you flip the logic here to reduce the nesting and throw if it isn't these two errors?